xonsh/tools.py

```diff
@@ -205,6 +216,28 @@ def __eq__(self, other):
         return all(map(operator.eq, self, other))
 
 
+class EnvPathDir(EnvPath):
```

Comment:
you're right, maybe we must solve this on a higher level as in when traversing the $PATH looking for commands [here](https://github.com/xonsh/xonsh/blob/master/xonsh/commands_cache.py#L97), it may also need a small cleanup/fixing. We either need to change this in `commands_cache.py` or change how we represent the `BASH_COMPLETION` envvar, maybe @gforsyth can chime in on that decision?